acra-censor/common/file_log_storage.go

```diff
@@ -20,25 +19,18 @@ func NewFileLogStorage(filePath string) (*FileLogStorage, error) {
 	}
 	fileLogStorage := &FileLogStorage{}
 	fileLogStorage.file = openedFile
-	fileLogStorage.filePath = filePath
-	fileLogStorage.openedToAppend = true
 	return fileLogStorage, nil
 }
 
 // ReadAll returns stored queries in raw form from internal file
 func (storage *FileLogStorage) ReadAll() ([]byte, error) {
-	return ioutil.ReadFile(storage.filePath)
+	return ioutil.ReadFile(storage.file.Name())
 }
 
 // WriteAll writes raw queries that need to be stored to internal file
 func (storage *FileLogStorage) WriteAll(p []byte) error {
-	if storage.openedToAppend {
-		openedFile, err := os.OpenFile(storage.filePath, os.O_TRUNC|os.O_RDWR|os.O_CREATE, 0600)
-		if err != nil {
-			return err
-		}
-		storage.file = openedFile
-		storage.openedToAppend = false
+	if err := storage.file.Truncate(0); err != nil {
```

Comment:
looks better)

Reply:
Sure. I just didn't know that Truncate method exists